libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -1054,15 +1056,17 @@ bool AP_GPS::get_lag(uint8_t instance, float &lag_sec) const
         // the user is always right !!
         return true;
     } else if (drivers[instance] == nullptr || state[instance].status == NO_GPS) {
-        // no GPS was detected in this instance so return a default delay of 1 measurement interval
-        lag_sec = 0.001f * (float)get_rate_ms(instance);
-        // check lack of GPS is consistent with user expectation
-        return state[instance].status == NO_GPS;
+        // no GPS was detected in this instance so return the worst possible lag term
+        if (_type[instance] == GPS_TYPE_NONE) {
+            lag_sec = 0.0f;
+            return true;
+        } else {
+            lag_sec = GPS_WORST_LAG_SEC;
+        }
+        return _type[instance] != GPS_TYPE_AUTO;
```

Comment:
Right so what this is aiming to do is state that:

If we have specified a GPS_TYPE (IE u-blox) then we will wait until that driver has been detected and loaded, and then set the optimal time. However as you observed auto detected instances may never show up. If I never return true from this function then the EKF will never boot. It's accepted that specifying a type is blocking on EKF booting till it's present, but auto will continue to boot. This is the change in behavior mentioned on the dev call.